openapi/componentsusers.schemas.yaml

```diff
@@ -0,0 +1,37 @@
+ComponentsUsers:
+  properties:
+    users:
+      type: array
+      items:
```

Comment:
Even though module yaml is a big improvement over having one huge yaml file, the yaml syntax is making me think that we should look at GRPC and pRPC again.

Reply:
Is there something that is missing in YAML syntax that the proto files from GRPC and pRPC files provide?

# Alternative 1: protobuf file - > Generated pRPC server and manually written frontend

Pros:
- pRPC servers works on App Engine
- Familiarity with proto files

Cons:
- pRPC does not generate frontend clients ([example](https://osscs.corp.google.com/chromium/infra/infra/+/main:appengine/monorail/static_src/shared/typedef.js;l=10-11?q=prpc%20-f:test&ss=chromium%2Finfra%2Finfra:appengine%2Fmonorail%2Fstatic_src%2F) Monorail has to manually write the typedefs for frontend. This is one thing that OpenAPI handles for us) 

# Alternative 2: protobuf file -> Generated GRPC server and GRPC frontend client

Pros:
- With [GRPC-Web](https://github.com/grpc/grpc-web) or [grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway), can get a generated frontend
- Familiarity with proto files

Cons:
- GRPC is not supported by App Engine. Could migrate to Cloud Run but that's a big shift in infrastructure.
- GRPC-Gateway requires additional infrastructure to have a proxy service


If I were to pick between the two options, I would pick option 2. That way we don't have to keep manually updating the client whenever we generate the backend in option 1. 

# My recommendation
Stay with this OpenAPI PR. This will allow developers to get better type checking automatically for frontend and backend. As a bonus, we can generate [protobuf](https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/protobuf-schema.md) files from OpenAPI as a quick migration path if want to move to alternative 2.

Reply:
> Is there something that is missing in YAML syntax that the proto files from GRPC and pRPC files provide?

The proto syntax is just more compact and easier to edit.  I suppose that I will get used to this syntax with time.